tests/conftest.py

```diff
@@ -0,0 +1,165 @@
+from sqlalchemy_utils import create_database
+from sqlalchemy_utils import database_exists
+from sqlalchemy_utils import drop_database
+
+from app import create_app
+from app import db
+from app.config import Config
+from app.config import RuntimeEnvironment
+from app.models import Host
+from app.utils import HostWrapper
+from tests.test_utils import get_required_headers
+from tests.test_utils import HOST_URL
+from tests.test_utils import inject_qs
+from tests.test_utils import SHARED_SECRET
+from tests.test_utils import validate_host
+
+
+@pytest.fixture(scope="session")
+def database():
+    config = Config(RuntimeEnvironment.server, "testing")
+    if not database_exists(config.db_uri):
+        create_database(config.db_uri)
+
+    yield
+
+    drop_database(config.db_uri)
+
+
+@pytest.fixture(scope="function")
+def flask_app(database):
+    app = create_app(config_name="testing")
+
+    # binds the app to the current context
+    with app.app_context() as ctx:
+        db.create_all()
+        ctx.push()
+
+        yield app
+
+        ctx.pop()
+        db.session.remove()
+        db.drop_all()
+
+
+@pytest.fixture(scope="function")
+def flask_client(flask_app):
+    flask_app.testing = True
+    return flask_app.test_client()
+
+
+@pytest.fixture(scope="function")
+def create_or_update_host(flask_client, subtests):
+    def _create(
+        host_data: Union[HostWrapper, List[HostWrapper]],
+        status=207,
+        host_status=None,
+        url=HOST_URL,
+        auth_type="account_number",
+        skip_host_validation=False,
+    ):
+        with subtests.test(msg="Creating or updating hosts"):
+            if not isinstance(host_data, List):
+                host_data = [host_data]
+
+            payload = [item.data() if isinstance(item, HostWrapper) else item for item in host_data]
+
+            response = flask_client.post(url, data=json.dumps(payload), headers=get_required_headers(auth_type))
+            assert response.status_code == status
+
+            host_response = json.loads(response.data)
+
+            if skip_host_validation or response.status_code > 299:
+                return host_response
+
+            host_status = host_status or [201]
+            if not isinstance(host_status, List):
+                host_status = [host_status]
+
+            for _key, _host in enumerate(host_data):
+                _host = _host if isinstance(_host, HostWrapper) else HostWrapper(_host)
+                assert host_response["data"][_key]["status"] == host_status[_key]
+                if host_response["data"][_key]["status"] < 299:
+                    validate_host(host_response["data"][_key]["host"], _host)
+
+            if len(payload) == 1:
+                return host_response["data"][0]["host"]
+            return [_host["host"] for _host in host_response["data"]]
+
+    return _create
+
+
+@pytest.fixture(scope="function")
+def get_host(flask_client, subtests):
+    def _get(url, status=200, **query_parameters):
```

Comment:
```suggestion
    def _get(url, expected_status=200, **query_parameters):
```

- Can we accept the parameters as a _dict_?

   ```suggestion
       def _get(url, expected_status=200, query_parameters):
   ```

- Should we use this function for all the various GET operations? We can rather create separate functions without the URL parameter or taking a collection of IDs.